### PR TITLE
LT-4061: CandlesHistory: add api to fetch low/high/eod mid prices for each period

### PR DESCRIPTION
**Pull request by lykke-vashetsin**

### CODE REVIEW

File: src/Lykke.Service.CandleHistory.Repositories/Candles/CandlesHistoryRepository.cs [hunk 1]
```diff
@@ -49,6 +49,12 @@ public async Task<IEnumerable<ICandle>> GetCandlesAsync(string assetPairId, Cand
                 throw;
             }
         }
+        
+        public async Task<(decimal? firstEod, decimal? lowest, decimal? highest)> GetPriceEvolutions(string assetPairId,
+            CandlePriceType priceType, CandleTimeInterval interval, DateTime? startDate)
+        {
+            throw new NotImplementedException();
```
Comment (anokhin-lkk) [inline src/Lykke.Service.CandleHistory.Repositories/Candles/CandlesHistoryRepository.cs:56]: Why not implemented?

**Review by anokhin-lkk — CHANGES_REQUESTED** [review]
Let's discuss this task

File: src/Lykke.Service.CandleHistory.Repositories/Candles/CandlesHistoryRepository.cs [hunk 2]
```diff
@@ -50,6 +50,17 @@ public async Task<IEnumerable<ICandle>> GetCandlesAsync(string assetPairId, Cand
             }
         }
 
+        public Task<decimal?> GetPricesEvolution(string assetPairId, CandlePriceType priceType, DateTime? startDate)
+        {
+            throw new NotImplementedException();
```
Comment (tarurar) [inline src/Lykke.Service.CandleHistory.Repositories/Candles/CandlesHistoryRepository.cs:55]: Is it intentionally?

File: src/Lykke.Service.CandleHistory.Repositories/Candles/SqlAssetPairCandlesHistoryRepository.cs [hunk 3]
```diff
@@ -59,6 +60,47 @@ public SqlAssetPairCandlesHistoryRepository(string assetName, string connectionS
             }
         }
 
+        [CanBeNull]
```
Comment (tarurar) [inline src/Lykke.Service.CandleHistory.Repositories/Candles/SqlAssetPairCandlesHistoryRepository.cs:63]: I think `[ItemCanBeNull]` is more appropriate here.

File: src/Lykke.Service.CandleHistory.Repositories/Candles/SqlAssetPairCandlesHistoryRepository.cs [hunk 4]
```diff
@@ -59,6 +60,47 @@ public SqlAssetPairCandlesHistoryRepository(string assetName, string connectionS
...
+                    {
+                        priceTypeVar = priceType,
+                        startDateVar = startDate
+                    }, null, commandTimeout);
+                
+                return result.HasValue ? Convert.ToDecimal(result.Value) : (decimal?)null;
+            }
+        }
+
+        [CanBeNull]
```
Comment (tarurar) [inline src/Lykke.Service.CandleHistory.Repositories/Candles/SqlAssetPairCandlesHistoryRepository.cs:83]: Same here.

**Review by tarurar — APPROVED** [review]

File: src/Lykke.Service.CandleHistory.Repositories/Candles/CandlesHistoryRepository.cs (hunk 2 above)
Comment (lykke-vashetsin) [reply src/Lykke.Service.CandleHistory.Repositories/Candles/CandlesHistoryRepository.cs:55]: it is for azure which we don't use

File: src/Lykke.Service.CandleHistory.Repositories/Candles/SqlAssetPairCandlesHistoryRepository.cs (hunk 3 above)
Comment (lykke-vashetsin) [reply src/Lykke.Service.CandleHistory.Repositories/Candles/SqlAssetPairCandlesHistoryRepository.cs]: deleted at all, not sure how it comes here :)

**Review by tarurar — APPROVED** [review]

**Pull request merged**